kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -410,20 +412,29 @@ def kompilation_up_to_date() -> bool:
 
         return old_digest == kompilation_digest()
 
+    def kompile_haskell() -> None:
+        _LOGGER.info(f'Kompiling definition: {foundry_main_file}')
+        _kompile(foundry_definition_dir, KompileTarget.HASKELL)
+
+    def kompile_llvm() -> None:
+        _LOGGER.info(f'Kompiling definition to LLVM dynamic library: {foundry_main_file}')
+        _kompile(
+            foundry_llvm_dir,
+            KompileTarget.LLVM,
+            llvm_kompile_type=LLVMKompileType.C,
+        )
+
     def update_kompilation_digest() -> None:
         digest_file = foundry_definition_dir / 'digest'
         digest_file.write_text(kompilation_digest())
 
     if not kompilation_up_to_date() or rekompile or not kompiled_timestamp.exists():
-        _LOGGER.info(f'Kompiling definition: {foundry_main_file}')
-        _kompile(foundry_definition_dir, KompileTarget.HASKELL)
-        if llvm_library:
-            _LOGGER.info(f'Kompiling definition to LLVM dy.lib: {foundry_main_file}')
-            _kompile(
-                foundry_llvm_dir,
-                KompileTarget.LLVM,
-                llvm_kompile_type=LLVMKompileType.C,
-            )
+        with concurrent.futures.ThreadPoolExecutor(max_workers=2) as executor:
```

Comment:
Posting it here as well so that it's easier to unfold the conversation:

> How much faster is it this way? I'd expect the resulting execution time to be approximately the same as the slower of the two executions.

Reply:
@tothtamas28 
 - with `--with-llvm-library` flag in parallel:  6:21.98
 - with `--with-llvm-library` flag on master: 8:40.81
 - without `--with-llvm-library` flag on master: 6:16.03

Reply:
Looks good, thanks!